manifests/install.pp

```diff
@@ -107,10 +107,12 @@
     }
     'RedHat': {
       if $telegraf::manage_repo {
-        if $facts['os']['name'] == 'Amazon' {
-          $_baseurl = "https://repos.influxdata.com/rhel/6/\$basearch/${telegraf::repo_type}"
+        if $facts['os']['name'] == 'Amazon' and $facts['os']['release']['major'] == '2023' {
```

Comment:
Since the first two conditions are Amazon I wonder if a nested `case` would be easier on the eyes:
```
if $facts['os']['name'] == 'Amazon' {
  case $facts['os']['release']['major'] {
    '2023': {
      $_baseurl = "${telegraf::repo_location}rhel/9/\$basearch/${telegraf::repo_type}"
    }
    default: {
      $_baseurl = "${telegraf::repo_location}rhel/6/\$basearch/${telegraf::repo_type}"
    }
  }
} else { ....
```
Just an idea..